### PR TITLE
Add DuckDB struct test and row as alias

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Add support for DuckDB-like syntax and functions to facilitate testing

### CODE REVIEW

**Comment by alamb** [comment]
wasmtest Ci failure has been fixed in https://github.com/apache/datafusion/pull/12844

File: datafusion/sqllogictest/test_files/struct.slt [hunk 1]
```diff
@@ -373,3 +373,52 @@ You reached the bottom!
 
 statement ok
 drop view complex_view;
+
+# Test row alias
+
+query ?
+select row('a', 'b');
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/struct.slt:380]: BTW there is pretty sweet struct literal syntax like `{ c1: 'a', c2: 'b'}` as well

```sql
> select { c1: 'a', c2: 'b'};
+---------------------------------------------------------+
| named_struct(Utf8("c1"),Utf8("a"),Utf8("c2"),Utf8("b")) |
+---------------------------------------------------------+
| {c1: a, c2: b}                                          |
+---------------------------------------------------------+
1 row(s) fetched.
Elapsed 0.002 seconds.
```

**Review by alamb — APPROVED** [review]
😍  -- thank you @jayzhan211

File: datafusion/sqllogictest/test_files/struct.slt (hunk 1 above)
Comment (jayzhan211) [reply datafusion/sqllogictest/test_files/struct.slt:380]: In DuckDB, they don't have column name. Stored like a tuple.

```
D select row('red', 1);
┌──────────────────────────┐
│    main.row('red', 1)    │
│ struct(varchar, integer) │
├──────────────────────────┤
│ (red, 1)                 │
└──────────────────────────┘
```

**Comment by jayzhan211** [comment]
Thanks @alamb

**Pull request merged**